speech/grpc/transcribe_streaming.py

```diff
@@ -90,19 +87,18 @@ def _audio_data_generator(buff):
         yield b''.join(data)
 
 
-def _fill_buffer(audio_stream, buff, chunk):
+def _fill_buffer(audio_stream, buff, chunk, stoprequest):
     """Continuously collect data from the audio stream, into the buffer."""
     try:
-        while True:
+        while not stoprequest.is_set():
             buff.put(audio_stream.read(chunk))
     except IOError:
-        # This happens when the stream is closed. Signal that we're done.
-        buff.put(None)
+        pass
```

Comment:
Without this, you'll have a race condition. ie if `_audio_data_generator`'s `buff.get()` is waiting for data in the buffer when `stoprequest` is triggered, it could potentially wait forever and never yield a value.